community/kernel/src/main/java/org/neo4j/kernel/api/bolt/SessionManager.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.bolt;
+
+import java.util.Set;
+
+public interface SessionManager
+{
+    void sessionActivated( KillableUserSession session );
+
+    void sessionHalted( KillableUserSession session );
```

Comment:
What's the difference between inactive/halted/killed? There are quite a few different adjectives in play here and it would be helpful to introduce some consistency of wording. For example, Can one deactivate something killable? Can one halt something that is active?